From 6e909b0b719155c026dc7117afc3059507c8167f Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Fri, 19 Oct 2018 15:50:01 -0700
Subject: [PATCH] TRITON-893 sdcadm check-health should ignore cainstsvc errors
 during transition to remove CA Reviewed by: Todd Whiteman
 <todd.whiteman@joyent.com> Approved by: Todd Whiteman
 <todd.whiteman@joyent.com>

---
 lib/sdcadm.js | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index 7952262..a466f20 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -3872,6 +3872,13 @@ SdcAdm.prototype.checkHealth = function checkHealth(opts, cb) {
                     return false;
                 }
 
+                // CA (cloud analytics) is being removed from core Triton in
+                // TRITON-884. During that transition, the "cabase" and
+                // "cainstsvc" agents on a CN are ignored for health checking.
+                if (inst.service === 'cabase' || inst.service === 'cainstsvc') {
+                    return false;
+                }
+
                 return true;
             });
 
@@ -3907,9 +3914,8 @@ SdcAdm.prototype.checkHealth = function checkHealth(opts, cb) {
             return;
         }
 
-        // there are a couple agents which don't actually have SMF services,
-        // so skip them
-        if (inst.service.match(/(agents_core|cabase)$/)) {
+        // These agents don't have an SMF service to check.
+        if (inst.service === 'agents_core') {
             next();
             return;
         }
@@ -3992,9 +3998,8 @@ SdcAdm.prototype.checkHealth = function checkHealth(opts, cb) {
             return;
         }
 
-        // there are a couple agents which don't actually have SMF services,
-        // so skip them
-        if (inst.service.match(/(agents_core|cabase)$/)) {
+        // These agents don't have an SMF service to check.
+        if (inst.service === 'agents_core') {
             next();
             return;
         }
-- 
2.21.0

